src/interface/validation.rs

```diff
@@ -0,0 +1,284 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Validate an interface and the version change between two interfaces.
+
+use std::{cmp::Ordering, fmt::Display};
+
+use log::info;
+
+use super::{
+    error::{Error, ValidationError, VersionChangeError},
+    Interface,
+};
+
+/// A change in version of an interface.
+///
+/// This structure is used to validate that the new version of an interface is a valid successor of
+/// the previous version. The version cannot decrease, and they cannot be the same (not really a
+/// version change).
+///
+/// This validates only the version change, not the interface itself. For that, see [`Interface::validate_with`].
+#[derive(Debug, Clone, Copy)]
+pub struct VersionChange {
+    next_major: i32,
+    next_minor: i32,
+    prev_major: i32,
+    prev_minor: i32,
+}
+
+impl VersionChange {
+    /// Create a new version change from a new and previous interfaces.
+    pub fn try_new(next: &Interface, prev: &Interface) -> Result<Self, VersionChangeError> {
+        let change = Self {
+            next_major: next.get_version_minor(),
+            next_minor: next.get_version_minor(),
+            prev_major: prev.get_version_major(),
+            prev_minor: prev.get_version_minor(),
+        };
+
+        change.validate()?;
+
+        Ok(change)
+    }
+
+    /// Returns the previous version
+    pub fn previous(&self) -> (i32, i32) {
+        (self.prev_major, self.prev_minor)
+    }
+
+    /// Returns the next version
+    pub fn next(&self) -> (i32, i32) {
+        (self.next_major, self.next_minor)
+    }
+
+    /// Private method for a version change validation.
+    ///
+    /// Validate if the version change is valid.
+    pub fn validate(&self) -> Result<(), VersionChangeError> {
+        let majior = self.next_major.cmp(&self.prev_major);
+        let minor = self.next_minor.cmp(&self.prev_minor);
+
+        match (majior, minor) {
+            (Ordering::Less, _) => Err(VersionChangeError::MajorDecresed(*self)),
+            (Ordering::Equal, Ordering::Less) => Err(VersionChangeError::MinorDecresed(*self)),
+            (Ordering::Equal, Ordering::Equal) => Err(VersionChangeError::SameVersion(*self)),
+            _ => Ok(()),
+        }
+    }
+}
+
+impl Display for VersionChange {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}.{} -> {}.{}",
+            self.next_major, self.next_minor, self.prev_major, self.prev_minor
+        )
+    }
+}
+
+impl Interface {
+    /// Validate if an interface is valid
+    pub fn validate(&self) -> Result<(), Error> {
+        // TODO: add additional validation
+        if self.get_version_major() == 0 && self.get_version_minor() == 0 {
```

Comment:
This could also be written as

```suggestion
        let version = (self.get_version_major(), self.get_version_minor());
        if version == (0, 0) {
```

which in my opinion is a bit more readable

Reply:
Done